griptape/engines/image/image_generation_engine.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+from typing import Optional
+
+from attr import field, define
+
+from griptape.artifacts import ImageArtifact
+from griptape.drivers import BaseImageGenerationDriver
+from griptape.rules import Ruleset
+
+
+@define
+class ImageGenerationEngine:
```

Comment:
I know we went back and forth on this, I wonder if we should instead create a `BaseImageGenerationEngine` that has 4 subclasses:
- `TextToImageGenerationEngine`
- `ImageVariationGenerationEngine`
- `ImageInpaintingGenerationEngine`
- `ImageOutpaintingGenerationEngine`

I think this is more consistent with how it is implemented elsewhere, it allows us to keep a simple `run` method in each class, and aligns nicely with the Tasks. 

And can we improve the consistency of the naming? Maybe something like:
- `PromptImageGenerationEngine`
- `VariationImageGenerationEngine`
- `InpaintingImageGenerationEngine`
- `OutpaintingImageGenerationEngine`

Reply:
Spun off ask #525